aws/request/http_request.go

```diff
@@ -0,0 +1,22 @@
+// +build go1.5
+
+package request
+
+import "net/http"
+
+func copyHTTPRequest(r *http.Request) *http.Request {
+	return &http.Request{
+		URL:           r.URL,
+		Header:        r.Header,
+		Close:         r.Close,
+		Form:          r.Form,
+		PostForm:      r.PostForm,
+		Body:          r.Body,
+		MultipartForm: r.MultipartForm,
+		Host:          r.Host,
+		Method:        r.Method,
+		Proto:         r.Proto,
+		ContentLength: r.ContentLength,
+		Cancel:        r.Cancel,
```

Comment:
It is a pain to modify the code, because I cannot build and run in my forked repo. Can you please add the test? Thanks,